### DISCUSSION TITLE
How to test the datafetchers that returns a relay connection

### DISCUSSION THREAD

**Discussion by ghost** [Netflix/dgs-framework#863] [Q&A] [answered]
Hello everyone, I am trying to do the datafetcher tests for a query that the response is a relay connection, but the framework does not generate the ProjectionRoot file to make the projection in the query when doing the tests in the datafetcher.
Is this a bug? Is there any examples doing data fetcher tests using a relay connection anywhere? Because I couldn't find any examples. Or I am forced to use the plain graphql string query in this type of queries using relay connection pagination?

Thank you.

**Comment by srinivasankavitha** [comment, marked as answer]
Hi - this is a gap in functionality since codegen does not recognize the @connection directive. So far, only the framework processes it to dynamically generate the connection types. We will need to add this capability, will do that soon. In the meantime, please use a graphql query string for the tests.

**Reply by ghost** [reply]
Thanks for the answer
I have explained a little bit in this issue about the problems I having now with testing the expected result, doing the query with a string now works, but testing the result and comparing it to my expected one to test the contract is very difficult (maybe I am missing something)

https://github.com/Netflix/dgs-codegen/issues/325

Thanks!
